src/databricks/labs/ucx/framework/tasks.py

```diff
@@ -63,6 +64,7 @@ def wrapper(*args, **kwargs):
             notebook=notebook,
             dashboard=dashboard,
         )
+        _TASK_ORDER.append(func.__name__)
```

Comment:
Why do we need to introduce a separate data structure, that has more potential for getting out of sync due to a bug?

Why not a field in the Task class, that holds the definition order (and sorting the tasks before the readme generation) or a topological sort based on depends_on field?

Reply:
The latter will not help with the steps ordering, only task ordering. I am open for suggestions on how to do this better.

Reply:
How about an extra field that holds a generated ID, which is starting at 1 and gets increased for every task added. Then we can sort on that field? @nfx

Reply:
> How about an extra field that holds a generated ID, which is starting at 1 and gets increased for every task added. Then we can sort on that field? 

that's exactly what I'm asking for :) but start with 0, to be consistent with array indexing